libraries/tlv-account-resolution/src/state.rs

```diff
@@ -149,7 +149,12 @@ impl ExtraAccountMetas {
         Self::add_to_vec::<T>(&mut instruction.accounts, data)
     }
 
-    /// Add the additional account metas and account infos for a CPI
+    /// Add the additional account metas and account infos for a CPI, while
+    /// de-escalating repeated accounts.
+    ///
+    /// If an added account already exists in the instruction with lower
+    /// privileges, match it to the existing account. This prevents a lower
+    /// program from gaining unexpected privileges.
```

Comment:
Do we also want to add this feature to the above `add_to_instruction` function?

Reply:
That's a good point, it would make things consistent. done!